### PR TITLE
Fix Notifications Overlay that opens automatically

### LINKED ISSUE #6130: Notifications Overlay Opens Automatically
#### Summary

The notifications overlay is opening automatically and after it has opened one time before that. This can get pretty annoying for users who have opened the notifications to clear them, but are still working on the same .

#### Expected vs Current Behavior
The notifications overlay should only open when a user clicks on it from the status bar. However, I am observing that it opens up automatically when a user has opened it once in the past. 

#### Steps to Reproduce

1. Do an action that produces a notification. (For example, edit a new or current object, and then save). 
2. Click the notification in the status bar to open the overlay. 
3. Close the notification overlay. 
4. Do another action that produces a notification (Again, you can edit the same object, create a new one, or navigate to another object and edit that). 
5. When you press save, notice that the notifications overlay opens up automatically. 

#### Environment

* Open MCT Version: R 2.1.5
* Deployment Type:  Dev
* OS: 
* Browser: Chrome

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

#### Additional Information

https://user-images.githubusercontent.com/48999852/213029796-3f79d67b-b32a-4311-a559-64d11e2b8a79.mov

### PR DESCRIPTION
**Pull request by 360macky**
Closes #6130.

### Problem

**TL;DR** The close button does not close properly because of the parent component

We have this current user flow:

1. User receives information notification ✅ Alert is displayed
2. User opens notification list ✅ Notification list is displayed
3. User closes notification list ⚠️

⚠️   At this point two things can happen: if the user closes the notification list before 3000 milliseconds (since the alert is sent) everything works correctly. The [variable in charge of displaying the NotificationList component](https://github.com/nasa/openmct/blob/d5adaf6e8c17b6e34fa0fa433ece7799d9d8f7b0/src/plugins/notificationIndicator/components/NotificationIndicator.vue#L44) will switch from `true` to `false`.

But after that time, if the user tries to close it, the error occurs.

The emitting event [to close of the NotificationsList component](https://github.com/nasa/openmct/blob/d5adaf6e8c17b6e34fa0fa433ece7799d9d8f7b0/src/plugins/notificationIndicator/components/NotificationsList.vue#L61) is not sent to the NotificationIndicator once it is hidden (not rendered).

Since the variable in charge of displaying the NotificationList component is still `true`, the next time Open MCT tries to send any kind of alert, it will open the alert with the notification list. Here's the bug.

### Solution

Adding `showNotificationsOverlay` to this condition continue to display the NotificationIndicator element even if there are no notifications. This allows the notification list closing flow to continue on its way without interruption.

```js
// If there are no notifications, is the notification list displayed? If so, keep the parent component rendered so the user can close it properly, otherwise just if there are notifications.
notifications.length === 0 ? showNotificationsOverlay : notifications.length > 0
```

This has minimal impact on the user experience, because the modal still hides all page content including the indicator itself.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by unlikelyzero** [comment]
This looks amazing! Would you mind adding a regression test for this area?

**Comment by 360macky** [comment]
> This looks amazing! Would you mind adding a regression test for this area?

Sure! I will push the test for your review.

File: e2e/tests/functional/notificationOverlay.e2e.spec.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        await createDomainObjectWithDefaults(page, { type: 'Display Layout' });
+
+        // Click on the button "Review 1 Notification"
+        await page.click('button[aria-label="Review 1 Notification"]');
+
+        // Verify that Notification List is open
+        expect(await page.locator('div[role="dialog"]').isVisible()).toBe(true);
+
+        // Wait until there is no Notification Banner
+        await page.waitForSelector('.c-message-banner__message', { state: 'detached'});
```
Comment (unlikelyzero) [inline e2e/tests/functional/notificationOverlay.e2e.spec.js:50]: Is there a more accessible locator available for the notification banner?

File: e2e/tests/functional/notificationOverlay.e2e.spec.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ *****************************************************************************/
+
+/*
+  This is a regression test to assure that the notification list behaves as expected.
+*/
+
+const { test, expect } = require('../../pluginFixtures');
+const { createDomainObjectWithDefaults } = require('../../appActions');
+
+test.describe('Notification Overlay Regression Test', () => {
```
Comment (ozyx) [inline e2e/tests/functional/notificationOverlay.e2e.spec.js:30]: This is looking awesome! Can we move this test to `e2e/tests/functional/notifications.e2e.spec.js` under a new test suite called "Notification Overlay"?

**Review by ozyx — CHANGES_REQUESTED** [review]

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
Test looks great! I just had a single question about accessible locators.

Would you mind moving the test into the existing notification.e2e.spec.js file?

File: e2e/tests/functional/notificationOverlay.e2e.spec.js (hunk 1 above)
Comment (360macky) [reply e2e/tests/functional/notificationOverlay.e2e.spec.js:50]: Sure, I'll replace it.

File: e2e/tests/functional/notificationOverlay.e2e.spec.js (hunk 2 above)
Comment (360macky) [reply e2e/tests/functional/notificationOverlay.e2e.spec.js:30]: Thanks. Yes, I'll move it.

**Review by unlikelyzero — APPROVED** [review]

File: e2e/tests/functional/notification.e2e.spec.js [hunk 3]
```diff
@@ -37,3 +38,42 @@ test.describe('Notifications List', () => {
         // Verify that the other notifications are still present in the notifications list
     });
 });
+
+test.describe('Notification Overlay Regression Test', () => {
```
Comment (ozyx) [inline e2e/tests/functional/notification.e2e.spec.js:42]: Can we rename this "Notification Overlay" ?

**Review by ozyx — CHANGES_REQUESTED** [review]
Looks great! Just have one minor request and this is good to merge.

File: e2e/tests/functional/notification.e2e.spec.js (hunk 3 above)
Comment (360macky) [reply e2e/tests/functional/notification.e2e.spec.js:42]: Sure!

**Review by ozyx — APPROVED** [review]
Awesome work. Really appreciate the quick turnaround on this! Thank you again.
